test/built-ins/TypedArray/length-invalid.js

```diff
@@ -0,0 +1,22 @@
+// Copyright (C) 2016 the V8 project authors. All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+/*---
+es6id: 22.2.1.2
+description: TypedArray rejects non-integer arguments
+info: >
+  22.2.1.2 %TypedArray% ( length )
+
+  ...
+  4. Let numberLength be ToNumber(length).
+  5. Let elementLength be ToLength(numberLength).
+  6. ReturnIfAbrupt(elementLength).
+  7. If SameValueZero(numberLength, elementLength) is false, throw a RangeError
+  exception.
+  ...
+features: [TypedArray]
+includes: [testTypedArray.js]
+---*/
+
+assert.throws(RangeError, function() {
```

Comment:
> I looked at the changes on the new specs but it's not published/released.

The change is present in the ES2016 2015-12-20 draft, [22.2.1.1, step 3](http://tc39.github.io/ecma262/#sec-typedarray):
> <ol><li value=3> If SameValue(NewTarget, here) is true, throw a TypeError exception.